python/cudf/cudf/core/dataframe.py

```diff
@@ -1700,10 +1697,32 @@ def _concat(cls, objs, axis=0, ignore_index=False):
             )
         else:
             index = Index._concat([o.index for o in objs])
-        data = [
-            (c, Series._concat([o[c] for o in objs], index=index))
-            for c in objs[0].columns
-        ]
+
+        # Currently we only support sort = False
+        # Change below when we want to support sort = True
+        # below functions as an ordered set
+
+        all_columns_list = []
+        all_columns_set = set()
+        for o in objs:
+            for col in o.columns:
+                if col not in all_columns_set:
+                    all_columns_list.append(col)
+                    all_columns_set.add(col)
```

Comment:
Can we do this with list / set comprehension instead? if you have a bunch of wide-ish dataframes this can actually be kinda expensive.
```
all_columns_list = [col for col in o.columns for o in objs]
all_columns_set = set(all_columns_list)
```

Reply:
We need to do this to maintain the order of the columns to match `pandas` behavior.  

I was looking for `ordered set` equivalent logic but could not find equivalent logic .
 Please let me know any suggestions you have for that implementation.

Reply:
If you need an ordered set you can just use an `OrderedDict` with the keys as `None`.